### PR TITLE
Add future{Comonad, Eq, Order, PartialOrder} to js

### PR DESCRIPTION
**Pull request by InTheNow**
Fix https://github.com/non/cats/issues/458

### CODE REVIEW

**Comment by InTheNow** [comment]
As I'm not sure where else to put these links, some references:

* [Event-based Concurrency (Advanced)](http://pages.cs.wisc.edu/~remzi/OSTEP/threads-events.pdf)
* [How JavaScript Timers Work](http://ejohn.org/blog/how-javascript-timers-work/)

**Comment by non** [comment]
Looks good to me. :+1:

File: js/src/main/scala/cats/jvm/std/future.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+trait FutureInstances0 extends FutureInstances1 {
+  def futureComonad(atMost: FiniteDuration)(implicit ec: E): Comonad[Future] =
+    new FutureCoflatMap with Comonad[Future] {
+      def extract[A](x: Future[A]): A =
+        Await.result(x, atMost)
+    }
+
+  def futureOrder[A: Order](atMost: FiniteDuration)(implicit ec: E): Eq[Future[A]] =
```
Comment (ceedubs) [inline js/src/main/scala/cats/jvm/std/future.scala]: I think this return type should be `Order` instead of `Eq`.

File: build.sbt [hunk 2]
```diff
@@ -216,11 +217,11 @@ lazy val publishSettings = Seq(
 ) ++ credentialSettings ++ sharedPublishSettings ++ sharedReleaseProcess 
 
 // These aliases serialise the build for the benefit of Travis-CI.
-addCommandAlias("buildJVM", ";macrosJVM/compile;coreJVM/compile;freeJVM/compile;freeJVM/test;stateJVM/compile;stateJVM/test;lawsJVM/compile;testsJVM/test;bench/test")
+addCommandAlias("buildJVM", ";macrosJVM/compile;coreJVM/compile;freeJVM/compile;freeJVM/test;stateJVM/compile;stateJVM/test;lawsJVM/compile;testsJVM/test;jvm/test;bench/test")
```
Comment (ceedubs) [inline build.sbt]: What's the difference between `jvm/test` and `testsJVM/test`? Similarly below for `js`.

File: js/src/test/scala/cats/tests/FutureTests.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+  implicit val throwableEq: Eq[Throwable] =
+    Eq.fromUniversalEquals
+
+  implicit val comonad: Comonad[Future] = futureComonad(timeout)
+
+  // Need non-fatal Throwables for Future recoverWith/handleError
+  implicit val nonFatalArbitrary: Arbitrary[Throwable] =
+    Arbitrary(arbitrary[Exception].map(_.asInstanceOf[Throwable]))
```
Comment (ceedubs) [inline js/src/test/scala/cats/tests/FutureTests.scala]: `asInstanceOf` shouldn't be necessary here. I think it should compile if you just use `.map(identity)`.

**Comment by ceedubs** [comment]
I left a few minor comments/questions, but in general this looks good to me.

File: build.sbt (hunk 2 above)
Comment (InTheNow) [inline build.sbt]: `testsJVM` and `testsJS` are the projects for the `tests` cross project.So neither contain actual code

`jvm` and `js` are top level projects that are _not_ cross projects, and do contain code

File: js/src/main/scala/cats/jvm/std/future.scala (hunk 1 above)
Comment (InTheNow) [inline js/src/main/scala/cats/jvm/std/future.scala]: Perhaps - but so should [the lvm version](https://github.com/non/cats/blob/master/jvm/src/main/scala/cats/jvm/std/future.scala#L2)

File: build.sbt (hunk 2 above)
Comment (ceedubs) [inline build.sbt]: Ah okay I think I understand now. Thanks for the explanation.

File: js/src/test/scala/cats/tests/FutureTests.scala (hunk 3 above)
Comment (InTheNow) [inline js/src/test/scala/cats/tests/FutureTests.scala]: ditto ^^  Perhaps - but so should [the lvm version](https://github.com/non/cats/blob/master/jvm/src/test/scala/cats/tests/FutureTests.scala)

File: js/src/main/scala/cats/jvm/std/future.scala (hunk 1 above)
Comment (InTheNow) [inline js/src/main/scala/cats/jvm/std/future.scala]: I've also noticed that the dir is wrong (has jvm in path). 

`js/src/main/scala/cats/jvm/std/future.scala` to `js/src/main/scala/cats/js/std/future.scala`

**Comment by InTheNow** [comment]
Shall I fix `jvm` version in this PR (easy) or a new one (strictly correct)?

**Comment by ceedubs** [comment]
I say throw the change in here. It doesn't bother me, and I have a suspicion that you'll still get a :+1: from @non :)

**Comment by non** [comment]
@ceedubs is correct :+1:

**Comment by InTheNow** [comment]
Cool, done already, ready for review

File: build.sbt (hunk 2 above)
Comment (non) [inline build.sbt]: Yeah, I am actually the origin of this confusing difference, but I think in the long-run it will make sense ;)

**Comment by non** [comment]
Looks good to me. Thanks @InTheNow!

**Comment by ceedubs** [comment]
:+1: thanks!

**Pull request merged**